src/patchedRequest.js

```diff
@@ -31,7 +31,9 @@ const oldPrepareRequestOptions = request.prepareRequestOptions
 request.prepareRequestOptions = function(requestOptions, methodOptions) {
 	// add our cancelToken support
 	if (methodOptions.cancelToken && typeof methodOptions.cancelToken === 'object') {
-		requestOptions.cancelToken = Object.assign({}, requestOptions.cancelToken || {}, methodOptions.cancelToken)
```

Comment:
So in fact, we should always use the given methodOptions.cancelToken instead of requestOptions.cancelToken.
So let's just assign it here and we'll be able to keep all the prototyping now